packages/dd-trace/src/plugins/database.js

```diff
@@ -1,9 +1,36 @@
 'use strict'
 
 const StoragePlugin = require('./storage')
+const TextMapPropagator = require('../opentracing/propagation/text_map.js')
+const Config = require('../../src/config')
+const config = new Config()
+const textMap = new TextMapPropagator(config)
+const carrier = {}
 
 class DatabasePlugin extends StoragePlugin {
   static get operation () { return 'query' }
+
+  createDBMPropagationCommentService () {
+    const dddbs = encodeURIComponent(this.config.service)
```

Comment:
These variables will be re-encoded on every query, but their values will never change for the lifetime of the process so their values should be cached to avoid the cost of encoding each variable each time.

Reply:
Or alternatively the resulting string including all variables could be cached but that would be more difficult to implement. Either way should be fast enough.

Reply:
We can throw this at the top:

```js
let dbmComment = ''
```

And then in the fuction:

```js
if (dbmComment) return dbmComment
...
dbmComment = `...`
return dbmComment
```

Or, to remove the slight perf hit of an `if` statement at runtime, should we instead do a method swap?

Reply:
Dynamic config will allow changing the service name at runtime, so a simple cache without any checks wouldn't really be enough unfortunately. We would have to first deprecate setting these options programmatically. This is why I mentioned it would be more difficult to implement than caching the values directly.

Reply:
Is the service name the only part that can dynamically change at runtime?

Reply:
No it was just an example but all of the values can.

Reply:
In that case we should be able to compare the four input values (`this.config.service` and the three `this.tracer._*` values) against a locally cached version. If the values differ then recalculate the return value, cache it, and return it. Later, if the four values are equal, then return the cached return value.

Reply:
I have updated this to only encode on the instance that the tags have changed